File: Api/Models/CharacterProfile.cs
using System.ComponentModel.DataAnnotations;

namespace GearForgeXI.Models;

public class CharacterProfile
{
    [Key]
    public int Id { get; set; }

    [Required]
    public string UserId { get; set; } = null!;

    [Required]
    [MaxLength(15)] // FFXI character name limit
    public string CharacterName { get; set; } = null!;

    [Required]
    public Server Server { get; set; }

    // Race information
    [Required]
    public Race Race { get; set; }

    // Timestamps
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

    // Navigation properties
    public ApplicationUser User { get; set; } = null!;
    public ICollection<CharacterJob> CharacterJobs { get; set; } = new List<CharacterJob>();
}

public enum Race
{
    HumeMale = 1,
    HumeFemale = 2,
    ElvaanMale = 3,
    ElvaanFemale = 4,
    TarutaruMale = 5,
    TarutaruFemale = 6,
    MithraFemale = 7,
    GalkanMale = 8,
}

public enum Server
{
    Bahamut = 1,
    Shiva = 2,
    Phoenix = 3,
    Carbuncle = 4,
    Fenrir = 5,
    Sylph = 6,
    Valefor = 7,
    Leviathan = 8,
    Odin = 9,
    Quetzalcoatl = 10,
    Siren = 11,
    Ragnarok = 12,
    Cerberus = 13,
    Bismarck = 14,
    Lakshmi = 15,
    Asura = 16,
}

File: Client/src/models/CharacterProfile.ts
import type { Job } from './Job';

export interface CharacterProfile {
  id: number;
  characterName: string;
  server: string;
  race: string;
  createdAt: string;
  updatedAt: string;
  characterJobs: CharacterJob[];
}

export interface CharacterJob {
  id: number;
  job: Job;
  jobLevel: number;
  masterLevel: number;
  updatedAt: string;
}

export interface CreateCharacterProfileDto {
  characterName: string;
  server: number;
  race: number;
}

export interface UpdateCharacterProfileDto {
  characterName?: string;
  server?: number;
  race?: number;
}

export interface UpdateCharacterJobDto {
  jobId: number;
  jobLevel: number;
  masterLevel: number;
}

export const SERVERS = [
  { id: 1, name: 'Bahamut' },
  { id: 2, name: 'Shiva' },
  { id: 3, name: 'Phoenix' },
  { id: 4, name: 'Carbuncle' },
  { id: 5, name: 'Fenrir' },
  { id: 6, name: 'Sylph' },
  { id: 7, name: 'Valefor' },
  { id: 8, name: 'Leviathan' },
  { id: 9, name: 'Odin' },
  { id: 10, name: 'Quetzalcoatl' },
  { id: 11, name: 'Siren' },
  { id: 12, name: 'Ragnarok' },
  { id: 13, name: 'Cerberus' },
  { id: 14, name: 'Bismarck' },
  { id: 15, name: 'Lakshmi' },
  { id: 16, name: 'Asura' },
] as const;

export const RACES = [
  { id: 1, name: 'HumeMale', displayName: 'Hume Male' },
  { id: 2, name: 'HumeFemale', displayName: 'Hume Female' },
  { id: 3, name: 'ElvaanMale', displayName: 'Elvaan Male' },
  { id: 4, name: 'ElvaanFemale', displayName: 'Elvaan Female' },
  { id: 5, name: 'TarutaruMale', displayName: 'Tarutaru Male' },
  { id: 6, name: 'TarutaruFemale', displayName: 'Tarutaru Female' },
  { id: 7, name: 'MithraFemale', displayName: 'Mithra Female' },
  { id: 8, name: 'GalkanMale', displayName: 'Galkan Male' },
] as const;

export const getRaceDisplayName = (raceName: string): string => {
  const race = RACES.find(r => r.name === raceName);
  return race ? race.displayName : raceName;
};
